### DISCUSSION TITLE
Controller method - require IP (or IP range)

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#9005] [Ideas]
It would be useful if we hadny method for restricting controller specific IP - for example when we want controller only run by CRON script. Method could work similarly like for example `requireAdmin()` controller mothod, throwing exception when does not meets its criteria.

**Comment by Anubarak** [comment]
This actually already exists https://www.yiiframework.com/doc/guide/2.0/en/security-authorization

You can add access filters and insert [Ips](https://www.yiiframework.com/doc/api/2.0/yii-filters-accessrule#$ips-detail) 
The advantage of this is: you can even add this behaviour to other controllers, so you can even restrict access to other controllers with in way
